### PR TITLE
[yaml] Add UDF docs

### PR DESCRIPTION
**Pull request by Polber**
PR adds docs for UDF's to `yaml_mapping.md` and moves the UDF tests to `yaml_mapping_test.py`.

Requires go/beam-pr/28114 be merged first.

This should not be included as part of the 2.51.0 release.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [x] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by Polber** [comment]
R: @robertwb

File: sdks/python/apache_beam/yaml/readme_test.py [hunk 1]
```diff
@@ -143,6 +143,12 @@ def input_file(self, name, content):
   def input_csv(self):
     return self.input_file('input.csv', 'col1,col2,col3\nabc,1,2.5\n')
 
+  def input_udf(self):
+    return self.input_file(
+        'udf.py',
+        'def my_mapping(row):\n  return row.col2 > 0\n'
```
Comment (robertwb) [inline sdks/python/apache_beam/yaml/readme_test.py:149]: Maybe use triple quotes for this content so it's easier to read.

File: sdks/python/apache_beam/yaml/readme_test.py [hunk 2]
```diff
@@ -155,14 +161,27 @@ def __exit__(self, *args):
     self.tempdir.cleanup()
 
 
+def replace_mapping_fields(spec, arg_name, arg_value):
+  if arg_name != 'language' and arg_name != 'fields':
+    spec['config']['fields'] = {
+        k: {arg: arg_value if arg == arg_name else v[arg]
+            for arg in v} if isinstance(v, dict) else v
+        for k,
```
Comment (robertwb) [inline sdks/python/apache_beam/yaml/readme_test.py:169]: Yapf messes up on `k, v`. You can put in ()'s for better wrapping.

File: sdks/python/apache_beam/yaml/readme_test.py [hunk 3]
```diff
@@ -155,14 +161,27 @@ def __exit__(self, *args):
     self.tempdir.cleanup()
 
 
+def replace_mapping_fields(spec, arg_name, arg_value):
+  if arg_name != 'language' and arg_name != 'fields':
```
Comment (robertwb) [inline sdks/python/apache_beam/yaml/readme_test.py:165]: Can this be phrased positively? I don't know what `arg_name` could be here. Is `language/fields` ever actually passed?

File: sdks/python/apache_beam/yaml/readme_test.py [hunk 4]
```diff
@@ -233,9 +254,9 @@ def parse_test_methods(markdown_lines):
             if 'providers:' in yaml_pipeline:
               test_type = 'PARSE'
             yield test_name, create_test_method(
-                test_type,
-                test_name,
-                yaml_pipeline)
+              test_type,
```
Comment (robertwb) [inline sdks/python/apache_beam/yaml/readme_test.py:257]: This was correct before. Check your editor settings?

**Review by robertwb — COMMENTED** [review]
The docs look good. Will wait until the other changed is merged before merging this.

**Comment by damccorm** [comment]
@Polber what are next steps here? Can we clean up the conflicts and merge?

**Comment by Polber** [comment]
@damccorm I have another PR I want to get in before refactoring this. Should open that one today

**Comment by damccorm** [comment]
@Polber what is the plan for this PR?

**Pull request closed without merging**